mux_test.go

```diff
@@ -2803,6 +2803,48 @@ func TestSubrouterNotFound(t *testing.T) {
 	}
 }
 
+// testOptionsMiddleWare returns 200 on an OPTIONS request
+func testOptionsMiddleWare(inner http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.Method == http.MethodOptions {
+			w.WriteHeader(http.StatusOK)
+			return
+		}
+		inner.ServeHTTP(w, r)
+	})
+}
+
+// TestRouterOrder Should Pass whichever order route is defined
+func TestRouterOrder(t *testing.T) {
+	handler := func(w http.ResponseWriter, r *http.Request) {}
+	router := NewRouter()
+	router.Path("/a/b").Handler(http.HandlerFunc(handler)).Methods(http.MethodGet)
+	router.Path("/a/{a}").Handler(nil).Methods(http.MethodOptions)
+	router.Use(MiddlewareFunc(testOptionsMiddleWare))
+
+	w := NewRecorder()
+	req := newRequest(http.MethodOptions, "/a/b")
+
+	router.ServeHTTP(w, req)
+
+	if w.Code != http.StatusOK {
+		t.Fatalf("Expected status code 200 (got %d)", w.Code)
+	}
+
+	reversedPathRouter := NewRouter()
```

Comment:
Break these into subtests - e.g.
```
t.Run("Test handler ordering", func(t *testing.T) { 
   // test 
})

t.Run("Test reversed handler ordering", func(t *testing.T) { 
   // test 
})
```

I would also add additional test cases:

1. Test ordering where handlers have the _same_ method & path, but a different handler (expected: the first handler added is the one executed)
2. Same path, different methods (no path variables)
3. Same as #2, but with middleware

Reply:
Have made the listed changes